templates/partials/analytics/reporting.html

```diff
@@ -56,11 +56,38 @@ <h3 translate>analytics.reporting.incomplete</h3>
   </div>
 
   <div class="time-select" ng-show="district">
-    <a class="btn btn-link" ng-class="{ 'disabled': filters.time_unit === 'month' && filters.quantity === 1 }" ng-click="setTime({ time_unit: 'month', quantity: 1 })">Last 1 Month</a>
-    <a class="btn btn-link" ng-class="{ 'disabled': filters.time_unit === 'month' && filters.quantity === 3 }" ng-click="setTime({ time_unit: 'month', quantity: 3 })">Last 3 Months</a>
-    <a class="btn btn-link" ng-class="{ 'disabled': filters.time_unit === 'month' && filters.quantity === 6 }" ng-click="setTime({ time_unit: 'month', quantity: 6 })">Last 6 Months</a>
-    <a class="btn btn-link" ng-class="{ 'disabled': filters.time_unit === 'year' && filters.quantity === 1 }" ng-click="setTime({ time_unit: 'year', quantity: 1 })">Last 1 Year</a>
-    <a class="btn btn-link" ng-class="{ 'disabled': filters.time_unit === 'year' && filters.quantity === 5 }" ng-click="setTime({ time_unit: 'year', quantity: 5 })">Last 5 Years</a>
+    <a class="btn btn-link"
+      ng-class="{ 'disabled': filters.quantity === 1 }"
+      ng-click="setTimeQuantity(1)"
+      translate="analytics.reporting.last_time_unit"
+      translate-values="{ time_unit: '{{ filters.time_unit }}' }"></a>
+    <a class="btn btn-link"
+      ng-class="{ 'disabled': filters.quantity === 3 }"
+      ng-click="setTimeQuantity(3)"
+      translate="analytics.reporting.last_time_unit.plural"
+      translate-values="{ quantity: 3, time_unit: '{{ filters.time_unit }}' }"></a>
+    <a class="btn btn-link"
+      ng-class="{ 'disabled': filters.quantity === 6 }"
+      ng-click="setTimeQuantity(6)"
+      translate="analytics.reporting.last_time_unit.plural"
+      translate-values="{ quantity: 6, time_unit: '{{ filters.time_unit }}' }"></a>
+    <a class="btn btn-link"
+      ng-class="{ 'disabled': filters.quantity === 12 }"
+      ng-click="setTimeQuantity(12)"
+      translate="analytics.reporting.last_time_unit.plural"
+      translate-values="{ quantity: 12, time_unit: '{{ filters.time_unit }}' }"></a>
```

Comment:
Here the `filters.time_unit` won't be translated. You could try to use the translate directive, eg:

`translate-values="{ quantity: 12, time_unit: '{{ filters.time_unit | translate }}' }"`